### PR TITLE
Log exceptions in EventHubMetricsProvider, Handle SystemOutOfBounds array exception thrown in GetValidWorkerCount

### PR DESCRIPTION
**Pull request by chiangvincent**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Index out of bounds error:
All("ScaleControllerEvents")
| where PreciseTimeStamp > ago(4h)
| where Level == 2
| where * contains "bounds" and * contains "eventhub"
| take 1
| project Details

```
System.IndexOutOfRangeException : Index was outside the bounds of the array.
   at Microsoft.Azure.WebJobs.Extensions.EventHubs.Listeners.EventHubsTargetScaler.GetScaleResultInternal(TargetScalerContext context,Int64 eventCount,Int32 partitionCount)
   at async Microsoft.Azure.WebJobs.Extensions.EventHubs.Listeners.EventHubsTargetScaler.GetScaleResultAsync(TargetScalerContext context)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Host.Scale.ScaleManager.GetTargetScalersResultAsync(ScaleStatusContext context,IEnumerable`1 targetScalersToProcess) at D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Scale\ScaleManager.cs : 167
```

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs [hunk 1]
```diff
@@ -78,9 +78,9 @@ public async Task<EventHubsTriggerMetrics> GetMetricsAsync()
             {
                 checkpoints = await Task.WhenAll(checkpointTasks).ConfigureAwait(false);
             }
-            catch
+            catch (Exception e)
             {
-                // GetCheckpointsAsync would log
+                _logger.LogWarning($"Encountered an exception while checking EventHub '{_client.EventHubName}'. Error: {e.Message}");
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs:83]:
```suggestion
                _logger.LogWarning($"Encountered an exception while checking Event Hub metrics for '{_client.EventHubName}'. Error: {e.Message}");
```

We check an Event Hub for many things.  The context won't be clear without some mention of metrics or scaling.

Comment (chiangvincent) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs]: Done -- added comment

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs [hunk 2]
```diff
@@ -80,7 +80,7 @@ public async Task<EventHubsTriggerMetrics> GetMetricsAsync()
             }
             catch (Exception e)
             {
-                _logger.LogWarning($"Encountered an exception while checking EventHub '{_client.EventHubName}'. Error: {e.Message}");
+                _logger.LogWarning($"Encountered an exception while getting checkpoints for EventHub '{_client.EventHubName}' used for scaling. Error: {e.Message}");
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs:83]:
```suggestion
                _logger.LogWarning($"Encountered an exception while getting checkpoints for Event Hub '{_client.EventHubName}' used for scaling. Error: {e.Message}");
```

Gotta watch the branding!

Comment (chiangvincent) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubMetricsProvider.cs]: ah this was copied from a warning line above -- will change that too

**Review by jsquire — APPROVED** [review]

**Pull request merged**